autospec/infile_bb_parser.py

```diff
@@ -0,0 +1,202 @@
+                v = v.replace("${ROS_SPN}", bb_dict.get('NAME'))
+
+        bb_dict[k] = v
+
+
+def get_src_url(bb_dict):
+    if bb_dict.get('SRC_URI'):
+        bb_dict["URL"] = bb_dict.get("SRC_URI").split('\\')[0].split(';')[0]
+
+
+def clean_values(value):
+    # remove beginning and end quote
+    if value.startswith('"') and value.endswith('"'):
+        value = value[1:-1]
+
+    # remove line continuations
+    value = "".join(value.split("\\"))
+    return value
+
+
+def update_inherit(line, bb_dict):
+    if 'inherits' in bb_dict:
+        bb_dict['inherits'].append(' '.join(line.split(' ', 1)[1:]))
+    else:
+        bb_dict['inherits'] = line.split(' ', 1)[1:]
+
+
+def read_in_command(line, depth):
+    for c in line:
+        if c == '{':
+            depth += 1
+        if c == '}' and depth > 0:
+            depth -= 1
+        if c != "\n":
+            continue
+    return depth
+
+
+def pattern_match_line(line):
+
+    operators = ["??=", "?=", ":=", "+=",
+                 "=+", ".=", "=.", "="]
+
+    for i, e in enumerate(operators):
+        operators[i] = '\\' + '\\'.join(e)
+
+    # Split line to be [Key, operator, value] if in operators list
+    oper_pattern = r"(^[A-Z]+[_\-${}\[\]A-Za-z0-9]*)\s(" + '|'.join(
+        operators) + r")\s(\".*\")"
+
+    return re.compile(oper_pattern).search(line)
+
+
+def evaluate_expr(op, cur_val, assignment):
+    if not cur_val:
+        return assignment
+    return operators_dict.get(op).format(cur_val, assignment)
+
+
+def write_to_dict(bb_dict, m):
+
+    if len(m.groups()) == 3:
+        key = m.group(1)
+        value = clean_values(m.group(3))
+        op = m.group(2)
+
+        #
+        # ??= is the weakest variable assignment, so if that variable already
+        # has an assignment, do not overwrite it. = has the highest precedence
+        # and ?= is between the two.
+        #
+        if key in bb_dict and op == "??=":
+            return bb_dict
+        elif key in bb_dict and op == "?=":
+            if not isinstance(bb_dict[key], list):
+                return bb_dict
+
+        if key in bb_dict and isinstance(bb_dict[key], list):
+            v = bb_dict[key]
+            del [v[1]]
+            bb_dict[key] = "".join(v)
+        try:
+            bb_dict[key] = evaluate_expr(op, bb_dict.get(key), value)
+            if op == "??=":
+                bb_dict[key] = [bb_dict[key], 1]
+        except AttributeError as error:
+            print("Missing operation functionality: {}".format(error))
+
+    return bb_dict
+
+
+def bb_scraper(bb_fp, bb_dict):
+    todo = []
+
+    if bb_dict.get('filename'):
+        bb_dict = scrape_version(bb_dict)
+
+    for line in bb_fp:
+        cont = None
+        if line.strip() and not line.strip().startswith('#'):
+            line = line.strip()
+
+            # If line is a command, create raw string of the command
+            # TODO: command could be python code
+            if not cont and line.startswith('do_'):
```

Comment:
I don't see where you ever set `cont` to anything but `None`, so that variable is basically a no-op.

Reply:
Done